examples/sandbox/index.ts

```diff
@@ -2518,139 +2508,132 @@ $ethAddr.on("click", async (e) => {
+        addressNList: hardenedPath.concat(relPath),
+        showDisplay: true,
+        address: result,
+      });
+      return result;
+    },
+  })
+);
+
+ethButtons.ethTx.on("click", (e) =>
+  performEthOperation(e, {
+    handler: async ({ tx }) => {
+      const res = await wallet.ethSignTx(tx);
+      return JSON.stringify(res);
+    },
+    useEIP1559: ethEIP1559Selected,
+  })
+);
+
+ethButtons.ethSign.on("click", (e) =>
+  performEthOperation(e, {
+    handler: async ({ hardenedPath, relPath }) => {
+      const result = await wallet.ethSignMessage({
+        addressNList: hardenedPath.concat(relPath),
+        message: "0x48656c6c6f20576f726c64", // "Hello World",
+      });
+      return `Address: ${result.address} Signature: ${result.signature}`;
+    },
+  })
+);
+
+ethButtons.ethSend.on("click", (e) =>
+  performEthOperation(e, {
+    handler: async ({ tx }) => {
+      if (wallet?.ethSendTx) {
+        const result = await wallet.ethSendTx(tx as core.ETHSignTx);
+        return result.hash;
+      } else {
+        const label = await wallet.getVendor();
+        return label + " does not support ethSendTx";
+      }
+    },
+    useEIP1559: ethEIP1559Selected,
+  })
+);
+
+ethButtons.ethVerify.on("click", (e) =>
+  performEthOperation(e, {
+    handler: async () => {
+      const signerAddress = "0x2068dD92B6690255553141Dfcf00dF308281f763";
+      const result = await wallet.ethVerifyMessage({
+        address: signerAddress,
+        message: "Hello World",
+        signature:
+          "0x61f1dda82e9c3800e960894396c9ce8164fd1526fccb136c71b88442405f7d09721725629915d10bc7cecfca2818fe76bc5816ed96a1b0cebee9b03b052980131b",
+      });
+      return result ? `✅ ${signerAddress} is the signer` : `❌ ${signerAddress} is NOT the signer`;
+    },
+  })
+);
 
-$ethEIP1559.on("click", async () => {
-  if (!ethEIP1559Selected) {
-    $ethEIP1559.attr("class", "button");
-  } else {
-    $ethEIP1559.attr("class", "button-outline");
-  }
+ethButtons.ethEIP1559.on("click", () => {
   ethEIP1559Selected = !ethEIP1559Selected;
+  ethButtons.ethEIP1559.attr("class", ethEIP1559Selected ? "button" : "button-outline");
 });
 
-$ethSignTypedData.on("click", async (e) => {
-  e.preventDefault();
-  if (!wallet) {
-    $ethResults.val("No wallet?");
-    return;
-  }
-  if (core.supportsETH(wallet)) {
-    const result = await wallet.ethSignTypedData({
-      typedData: eip712["calculateHashesOnDevice"]["typedData"],
-      addressNList: core.bip32ToAddressNList("m/44'/60'/0'/0/0"),
-    });
-    $ethResults.val(JSON.stringify(result, null, 2));
-  }
-});
+const doesRecoveredMatchAddress = (typedData: TypedData, fromAddress: string, signature: string): boolean => {
+  const recoveredAddress = sigUtil.recoverTypedSignature({
+    data: typedData,
+    signature: signature,
+    version: sigUtil.SignTypedDataVersion.V4,
+  });
+  return toChecksumAddress(recoveredAddress) === toChecksumAddress(fromAddress);
+};
```

Comment:
Added this utility method to verify the address from the signature. It was previously returning a wrong address before my fix in hdwallet-keepkey/src/ethereum.ts ethSignTypedData. This should help us immediately see any future regressions